src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -373,6 +373,8 @@ private async Task DelayThenInvokeDiagnosticsAsync(ScriptFile[] filesToAnalyze,
             {
                 ScriptFileMarker[] semanticMarkers = await AnalysisEngine.AnalyzeScriptAsync(scriptFile.Contents).ConfigureAwait(false);
 
+                // Clear old PSSA-results, but keeping parser-errors ("PowerShell"-source).
+                scriptFile.DiagnosticMarkers.RemoveAll(m => m.Source == "PSScriptAnalyzer");
```

Comment:
```suggestion
                // Clear existing PSScriptAnalyzer markers (but keep parser-errors where the source is "PowerShell")
                // so that they are not duplicated when re-opening files.
                scriptFile.DiagnosticMarkers.RemoveAll(m => m.Source == "PSScriptAnalyzer");
```